File: test-code/src/main/java/io/philo/test_code/app/PersonController.java
package io.philo.test_code.app;

import io.philo.test_code.app.dto.PersonCreateRequest;
import io.philo.test_code.app.dto.PersonCreateResponse;
import io.philo.test_code.app.dto.PersonListResponse;
import io.philo.test_code.app.dto.PersonListResponses;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RequestMapping("/person")
@RestController
@RequiredArgsConstructor
public class PersonController {

    private final PersonRepository repository;

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public PersonCreateResponse add(@RequestBody PersonCreateRequest request) {

        String name = request.name();
        Person person = new Person(name);
        repository.save(person);

        return new PersonCreateResponse(person.getId());
    }

    @GetMapping
    public PersonListResponses list() {

        List<Person> entities = repository.findAll();
        List<PersonListResponse> dtos = entities.stream()
                .map(PersonListResponse::new)
                .toList();

        return new PersonListResponses(dtos);
    }
}

File: common-jpa/src/main/kotlin/io/philo/presentation/PersonController.kt
package io.philo.presentation

import io.philo.entity.PersonEntity
import io.philo.repository.PersonRepository
import org.springframework.transaction.annotation.Transactional
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/person")
class PersonController(private val personRepository: PersonRepository) {

    @RequestMapping("/test-save")
    @Transactional
    fun testSave() {

        personRepository.save(PersonEntity(1, "foo_11"))

        personRepository.findAll()
    }

    @RequestMapping("/test-save-2")
    @Transactional
    fun testSave2() {

        personRepository.save(PersonEntity(1, "boo_22"))

        personRepository.findAll()
    }

    @RequestMapping("/test-get")
    @Transactional
    fun testGet() {

        personRepository.findAll()
    }
}
